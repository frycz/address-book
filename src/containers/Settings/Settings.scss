.settings {
    display: flex;
    justify-content: center;

    &__title, &__back-link {
        text-align: center;
    }
}
