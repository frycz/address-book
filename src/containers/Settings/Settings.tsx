import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { State } from "../../redux/store";
import { paths } from "../../router";
import { Countries } from "../../redux/store";
import { updateSettings } from '../../redux/book/actions'

interface Props {
  initialCountries: Countries;
  updateSettings: typeof updateSettings;
}

const Settings: React.FC<Props> = ({ initialCountries, updateSettings }) => {

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, checked } = e.target;
    updateSettings({
      ...initialCountries,
      [name]: checked
    })
  };

  return (
    <div className="settings">
      <p>Settings</p>
      {Object.entries(initialCountries).map(([country, isSelected]) => (
        <div key={country}>
          <input
            id={`settings-${country}`}
            type="checkbox"
            name={country}
            checked={isSelected}
            onChange={handleChange}
          />
          <label htmlFor={`settings-${country}`}>{country.toUpperCase()}</label>
        </div>
      ))}
      <Link to={paths.main}>Back to list</Link>
    </div>
  );
};

export default connect((state: State) => ({
  initialCountries: state.book.countries
}), {
  updateSettings
})(Settings);
