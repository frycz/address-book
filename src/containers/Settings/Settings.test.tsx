import React from "react";
import { shallow } from "enzyme";
import { Settings, Props } from "./Settings";

const mockProps: Props = {
  initialCountries: { ch: true, es: true, fr: false },
  updateSettings: jest.fn()
};

describe("Settings", () => {
  const settings = shallow(<Settings {...mockProps} />);

  it("should match smapshot", () => {
    expect(settings.debug()).toMatchSnapshot();
  });

  it("should render countries checkboxes", () => {
    expect(settings.find('input[type="checkbox"]')).toHaveLength(3);
  });

  it("should check ch", () => {
    expect(settings.find('#settings-ch').props().checked).toEqual(true);
  });

  it("should render check es", () => {
    expect(settings.find('#settings-es').props().checked).toEqual(true);
  });

  it("should not check fr", () => {
    expect(settings.find('#settings-fr').props().checked).toEqual(false);
  });
});
