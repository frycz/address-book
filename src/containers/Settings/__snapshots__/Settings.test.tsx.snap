// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings should match smapshot 1`] = `
"<div className=\\"settings\\">
  <div className=\\"settings__container\\">
    <p className=\\"settings__title\\">
      Settings
    </p>
    <div>
      <input id=\\"settings-ch\\" type=\\"checkbox\\" name=\\"ch\\" checked={true} onChange={[Function: handleChange]} />
      <label htmlFor=\\"settings-ch\\">
        CH
      </label>
    </div>
    <div>
      <input id=\\"settings-es\\" type=\\"checkbox\\" name=\\"es\\" checked={true} onChange={[Function: handleChange]} />
      <label htmlFor=\\"settings-es\\">
        ES
      </label>
    </div>
    <div>
      <input id=\\"settings-fr\\" type=\\"checkbox\\" name=\\"fr\\" checked={false} onChange={[Function: handleChange]} />
      <label htmlFor=\\"settings-fr\\">
        FR
      </label>
    </div>
    <p className=\\"settings__back-link\\">
      <Link to=\\"/\\">
        Back to list
      </Link>
    </p>
  </div>
</div>"
`;
